sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/component.py

```diff
@@ -110,6 +110,28 @@ def _serialize(self, obj, *, many: bool = False):
                 ret[attr_name] = self.get_attribute(obj, attr_name, None)
         return ret
 
+    @pre_load()
+    def convert_input_value_to_str(self, data: dict, **kwargs) -> dict:
+        """
+        Convert the non-str value in input to str.
+
+        When load the v1.5 component yaml, true/false will be converted to bool type and yyyy-mm-dd will be
+        converted to date type. In order to be consistent with before, it needs to be converted to str type.
+        """
+        def convert_to_str(value):
+            if isinstance(value, bool):
+                return str(value).lower()
```

Comment:
Why lower() here? Do we support both "true" and "True"?

Reply:
Because in smile the founded default value of inputs is true/false in lowercase, and confirmed with Clement that it is converted to lowercase

Reply:
sadly, Looks like we have such 'default:True' in office team's component

Reply:
But we have to pick a side with current implementation. May modify our migration doc on this behavior

Reply:
If really needed, we can use a different yaml loader to load "True" in string for internal components only.

If we have confirmed the lower behavior in Smile migration, we can merge this PR first, then please help open a task for me if we need to guarantee reuse in Shrike migration.